apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/site-logo/index.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Render site-logo block file.
+ *
+ * @package full-site-editing
+ */
+
+/**
+ * Renders a site-logo.
+ *
+ * @param array  $attributes Block attributes.
+ * @param string $content    Block content.
+ * @return string
+ */
+function render_site_logo( $attributes, $content ) {
+
+	if ( ! function_exists( 'get_custom_logo' ) || ! function_exists( 'has_custom_logo' ) ) {
+		return '';
+	}
+
+	if ( ! has_custom_logo() ) {
+		return '';
+	}
+
+	$add_site_link = $attributes['addSiteLink'];
+
+	//publish view
```

Comment:
Why is different handling needed for front-end and edit contexts here?

Reply:
Logos are wrapped with a link to the homepage. The edit view simply returns an image so folks aren't redirected out of the editing flow.

Reply:
Another option is to wrap the `ServerSideRender` component with a `Disabled` component which [disables all the pointer events](https://github.com/WordPress/gutenberg/blob/a57202e635ff0cd190a09128d27951620db9a870/packages/components/src/disabled/style.scss#L3).

This is what [Gutenberg does with the `video` block](https://github.com/WordPress/gutenberg/blob/a57202e635ff0cd190a09128d27951620db9a870/packages/block-library/src/video/edit.js#L284) to prevent the user from playing videos within the editor (something we replicated in the [VideoPress extension](https://github.com/Automattic/jetpack/blob/96a9b033147b723eb1e6f5052151dc01c4f3b165/extensions/blocks/videopress/edit.js#L144)).

Reply:
Going to move away from the dynamic block here following https://github.com/Automattic/wp-calypso/pull/33714#issuecomment-500495253